### PR TITLE
Update all-inclusive headers, and CMakeFiles for including all headers

### PR DESCRIPTION
**Pull request by antkmsft**

### CODE REVIEW

**Review by vhvb1989 — APPROVED** [review]
<3 amazing..
Is there a tool you use to find missing headers ?  Or is this all manual talent?

Thank you.

Please wait for @gearama  and @Jinming-Hu  approval before merging.

**Comment by antkmsft** [comment]
@vhvb1989, thank you! As they say, I had time and inspiration :) - this time I was in a specific stubborn mood so it even felt satisfying to review such things manually.
Hence I rearranged the order of some headers in CMakeLists, so it visually looks like the tree structure in the Solution Explorer, and it is easier to compare.

File: sdk/storage/azure-storage-blobs/CMakeLists.txt
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-blobs/CMakeLists.txt:61]: We keep the files in alphabetical order in storage.

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs.hpp
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs.hpp:16]: I won't expect people directly depend on these two files. But I assume it's ok to add them here.

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs.hpp
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs.hpp:23]: I don't think we should expose `rest_client.hpp`. It may be renamed in the future. I think we should only expose it indirectly, like via `blob_options.hpp`.

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs.hpp
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs.hpp:24]: We divide include headers into groups in storage, following [this](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes) rule. 

So please move this header to after `#pragma once` and make it a separate group.

File: sdk/storage/azure-storage-common/CMakeLists.txt
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-common/CMakeLists.txt:66]: also, alphabetical order

File: sdk/storage/azure-storage-blobs/CMakeLists.txt
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/CMakeLists.txt:61]: This is a header, why is it put in SOURCE ?

**Review by Jinming-Hu — COMMENTED** [review]
It basically looks good to me, except for a few small comments. 

So I have a question, if A depends on B, B depends on C. Do we need to include C in A?

File: sdk/keyvault/azure-security-keyvault-certificates/CHANGELOG.md
Comment (ahsonkhan) [inline sdk/keyvault/azure-security-keyvault-certificates/CHANGELOG.md:9]: What motivated this? For keyvault keys, our simplified header is called keyvault_keys.hpp.

File: sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keyvault_keys.hpp
Comment (ahsonkhan) [inline sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keyvault_keys.hpp:14]: If we really want to do this going forward, let's come up with a naming guideline to enforce this for future SDKs. We may want to leave already shipping SDKs like Keyvault Keys, as is. It's a small wart, imo, and not worth introducing multiple "top-level" header files that would require us to update samples. We now end up with multiple, which defeats the purpose and not worth the trade-off.

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs.hpp
Comment (antkmsft) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs.hpp:16]: It is easier to include all, rather than judge individually. The goal of the file is to include all public APIs. The headers are in the public include directory, so I think it is easier to include.

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs.hpp
Comment (antkmsft) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs.hpp:23]: If it gets renamed, the all-inclusive header should get updated too. As I said above, the goal is to include all public headers. If customer can see it intellisense, when typing `#include <azure/storage/blobs/`, it should be included in this file.

File: sdk/storage/azure-storage-common/CMakeLists.txt
Comment (antkmsft) [reply sdk/storage/azure-storage-common/CMakeLists.txt:66]: I copied the visual order of the expanded tree I see in Solution explorer, so it is easier to check if I missed something. If some tool does rearrange the order to strictly alphabetical, I am ok with it.

File: sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keyvault_keys.hpp
Comment (antkmsft) [reply sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keyvault_keys.hpp:14]: We do update all the samples to use new. The old one is quietly there, because we can't break it. I think it is the right way.

The rule should be, if the headers are in `azure/service/...` directory, the all-inclusive header is `azure/service.hpp`

File: sdk/keyvault/azure-security-keyvault-certificates/CHANGELOG.md
Comment (antkmsft) [reply sdk/keyvault/azure-security-keyvault-certificates/CHANGELOG.md:9]: If keyvault include files are in `azure/keyvault/certificates/`, then the all-inclusive file is `azure/keyvault/certificates.hpp`.

File: sdk/storage/azure-storage-blobs/CMakeLists.txt
Comment (antkmsft) [reply sdk/storage/azure-storage-blobs/CMakeLists.txt:61]: it is in `src` folder, i.e. not something discoverable by the code that has path to include directory. I don't think either thing makes a technical difference, but right now, it was useful for visual comparison against Solution Explorer.

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs.hpp
Comment (antkmsft) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs.hpp]: Done - here, and 3 other occurrences.

**Review by gearama — APPROVED** [review]
hope it builds :)

**Comment by antkmsft** [comment]
> So I have a question, if A depends on B, B depends on C. Do we need to include C in A?

Hey @Jinming-Hu, I view these files as such: if a file is present in `azure/service/...` (including subrirectories), it should be included in `azure/service.hpp`. Without too much analysis, because it is easier (excluding the headers in `azure/service/internal`).

I know, sometimes you can get away with just including `A` (because it should include `B`), but there could be confusion, if a user wants to use `B`, but for some reason (let's say `A` no longer includes it)  itis not available when they `#include <azure/service.hpp>`, so they need to write `#include <azure/service/b.hpp>`. That would be strange.

In general, I'd view it this way: as a library owner, feel free to have less files included if you know that `A` includes `B`. But it is much easier to verify and even automate, if it would include all files in the include directory (except `azure/service/internal`).
After all, it is harmless to include it explicitly, twice.

**Comment by antkmsft** [comment]
@Jinming-Hu, is it looking good for you, can I check this in?
I also had to lower code coverage a bit - it looks like some other change has sneaked past the CI, and the current coverage in Storage is now slightly lower. But I don't think it is related with anything in this PR (although even if it is, I think it rather should be fixed in a separate PR).

**Review by Jinming-Hu — APPROVED** [review]

**Pull request merged**
